internal/product/accessor_test.go

```diff
@@ -447,32 +447,36 @@ func Test_GetProductsByVendor(t *testing.T) {
 			ctx            = context.Background()
 			c              = setupProductAccessorTestComponent(t)
 			expectedResult = sqlmock.NewRows(productColumns).
-					AddRow(products[0].ID, "", "", "", "", products[0].Name, "", products[0].ModifiedDate, "").
-					AddRow(products[1].ID, "", "", "", "", products[1].Name, "", products[1].ModifiedDate, "")
+					AddRow(productVendors[0].ID, "", "", productVendors[0].Name, "", "", "", "", "", "", productVendors[0].ModifiedDate, "").
+					AddRow(productVendors[1].ID, "", "", productVendors[1].Name, "", "", "", "", "", "", productVendors[1].ModifiedDate, "")
 		)
 		defer c.db.Close()
 
-		query := getProductsByVendorQuery + " LIMIT $2 OFFSET $3"
+		query := getProductVendorsByVendorQuery + " LIMIT $2 OFFSET $3"
 		c.mock.ExpectQuery(query).
 			WithArgs(vendorID, args.Limit, args.Offset).
 			WillReturnRows(expectedResult)
 
 		totalRows := sqlmock.NewRows([]string{"count"}).AddRow(2)
-		countQuery := `SELECT COUNT(*) FROM product_vendor WHERE vendor_id=$1`
+		countQuery := `SELECT COUNT(*)
+			FROM product_vendor pv
+			JOIN price pr ON pr.product_vendor_id = pv.id
+			WHERE pr.vendor_id = $1
+		`
```

Comment:
I just noticed this query is wrong... What if the filter is on, the countQuery would still be the unfiltered count 🤦‍♂️